src/Azure.Functions.Cli/Common/DurableManager.cs

```diff
@@ -0,0 +1,362 @@
+        {
+            Initialize(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+
+            var historyString = await _orchestrationService.GetOrchestrationHistoryAsync(instanceId, null);
+
+            JArray history = JArray.Parse(historyString);
+
+            JArray chronological_history = new JArray(history.OrderBy(obj => (string)obj["TimeStamp"]));
+            foreach (JObject jobj in chronological_history)
+            {
+                // Convert EventType enum values to their equivalent string value
+                var parsed = Enum.TryParse(jobj["EventType"].ToString(), out EventType eventName);
+                jobj["EventType"] = eventName.ToString();
+            }
+
+            ColoredConsole.Write($"{chronological_history.ToString(Formatting.Indented)}");
+        }
+
+        public async Task GetInstances(string connectionStringKey, string taskHubName, DateTime createdTimeFrom, DateTime createdTimeTo, IEnumerable<OrchestrationStatus> statuses, int top, string continuationToken)
+        {
+            Initialize(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+
+            DurableStatusQueryResult queryResult = await _orchestrationService.GetOrchestrationStateAsync(createdTimeFrom, createdTimeTo, statuses, top, continuationToken);
+
+            // TODO? Status of each instance prints as an integer, rather than the string of the OrchestrationStatus enum
+            ColoredConsole.WriteLine(JsonConvert.SerializeObject(queryResult.OrchestrationState, Formatting.Indented));
+
+            ColoredConsole.WriteLine(Green($"Continuation token for next set of results: '{queryResult.ContinuationToken}'"));
+        }
+
+        public async Task GetRuntimeStatus(string connectionStringKey, string taskHubName, string instanceId, bool showInput, bool showOutput)
+        {
+            Initialize(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+
+            var status = (await _orchestrationService.GetOrchestrationStateAsync(instanceId, allExecutions: false, fetchInput: showInput)).FirstOrDefault();
+
+            if (status == null)
+            {
+                ColoredConsole.WriteLine(Red($"Could not find an orchestration instance with id '{instanceId}'"));
+            }
+            else
+            {
+                status.Output = (showOutput) ? status.Output : null;
+                ColoredConsole.WriteLine(JsonConvert.SerializeObject(status, Formatting.Indented));
+            }            
+        }
+
+         public async Task PurgeHistory(string connectionStringKey, string taskHubName, DateTime createdAfter, DateTime createdBefore, IEnumerable<OrchestrationStatus> runtimeStatuses)
+        {
+            Initialize(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+
+            var stats = await _orchestrationService.PurgeInstanceHistoryAsync(createdAfter, createdBefore, runtimeStatuses);
```

Comment:
Run through the IEnumerable passed in early (i.e. have a local and call `ToArray()` on it) and pass that to both methods using it below (`PurgeInstanceHistoryAsync` and `string.Join`)